### PR TITLE
Form

### PR DESCRIPTION
**Pull request by EllieChen-Git**
1. I installed 'bootswatch' to play with it. I didn't do anything important for this. You can ignore this.
2.  I decided to remain 'register page -> register form' & 'sign in page -> sign in form' structure (followed Garret's convention).
3. I created a folder 'toDelete(awaiting confirmation)' and I moved all the not-in-use file there. We can go through it on Monday to decide if we need them or not. If not, we'll just delete the whole folder.
4. I also tried to change all the 'href' to 'Link to', but I'm unable to change some of them. I'll explain in the comment below.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import ImageManagement from "./components/images_uploading/ImageManagement";
 import { setAuthToken, setUserInfo } from "./actions";
 import LocalApi from "./apis/LocalApi";
 import RegisterPage from "./components/pages/RegisterPage";
-import SigninForm from "./components/forms/SignInForm";
+import SignInPage from "./components/pages/SignInPage";
```
Comment (EllieChen-Git) [inline src/App.js:14]: Just try to maintain a more consistent structure here.

File: src/App.js [hunk 2]
```diff
@@ -54,10 +59,11 @@ class App extends Component {
         </BrowserRouter>
       );
     }
-    if (user.is_admin === true) {
+    if (user.is_admin) {
```
Comment (EllieChen-Git) [inline src/App.js:62]: Tried to clean code: 'if (user.is_admin === true) is the same as ' (if (user.is_admin))'

File: src/App.js [hunk 3]
```diff
@@ -71,16 +77,15 @@ class App extends Component {
           </Switch>
         </BrowserRouter>
       );
-    }
-    if (!user.is_admin) {
+    } else {
```
Comment (EllieChen-Git) [inline src/App.js:80]: Tried to clean code: 'if (!user.is_admin)' is the same as 'else'

File: src/components/AnnotationList.js [hunk 4]
```diff
@@ -22,12 +23,9 @@ class AnnotationList extends Component {
                   <div> Lot: {image.lot}</div>
                   <div> Unit Number: {image.unitNumber}</div>
                   <div> Product Description:{image.productDescription}</div>
-                  <Button
-                    variant="success"
-                    href={"annotations/" + annotation._id}
-                  >
-                    Go to Annotation
-                  </Button>
+                  <Link to={"annotations/" + annotation._id}>
```
Comment (EllieChen-Git) [inline src/components/AnnotationList.js:26]: Changed from 'href' to 'Link to'

File: src/components/forms/RegisterForm.js [hunk 5]
```diff
@@ -79,10 +80,9 @@ class RegisterForm extends Component {
           </Button>
         </Form>
         <h3>Already have an account?</h3>
-
-        <Button variant="success" href="/">
-          Sign In
-        </Button>
+        <Link to="/">
```
Comment (EllieChen-Git) [inline src/components/forms/RegisterForm.js:83]: Changed from 'href' to 'Link to'

File: src/components/forms/SignInForm.js [hunk 6]
```diff
@@ -3,25 +3,29 @@ import LocalApi from "../../apis/LocalApi";
...
   state = {
     username: "",
-    password: ""
+    password: "",
+    error: null
   };
 
   onFormSubmit = async event => {
     event.preventDefault();
+    const { setAuthToken } = this.props;
```
Comment (EllieChen-Git) [inline src/components/forms/SignInForm.js:17]: Desctructred 'setAuthToken' to clean code. (We destructured setAuthToken in 'resister form', so I tried to make the code more consistent. )

File: src/components/forms/SignInForm.js [hunk 7]
```diff
@@ -65,9 +69,9 @@ class SigninForm extends Component {
         </Form>
         <h3>Don't Have An Account?</h3>
 
-        <Button variant="danger" href="/register">
-          Register
-        </Button>
+        <Link to="/register">
```
Comment (EllieChen-Git) [inline src/components/forms/SignInForm.js:72]: Changed from 'href' to 'Link to'

File: src/components/images_uploading/DisplayImageList.js [hunk 8]
```diff
@@ -44,13 +44,41 @@ class UserAssignmentDropDown extends React.Component {
 }
 
 class DisplayImageList extends Component {
+  constructor(props) {
```
Comment (EllieChen-Git) [inline src/components/images_uploading/DisplayImageList.js:47]: Now we can 'delete apartment' and after we delete any apartment, the page will automatically refresh.
The code can definitely be improved, but at least it's working now (I'll see if I can make it better later T^T)

File: src/components/images_uploading/DisplayImageList.js [hunk 9]
```diff
@@ -63,11 +91,30 @@ class DisplayImageList extends Component {
               <UserAssignmentDropDown imageId={image._id} users={users} />
 
               <Button
+                style={{ marginLeft: 3 }}
                 variant="success"
                 href={`${process.env.REACT_APP_BASEURL}/images/${image.s3key}`}
               >
+                {/* this one can't be changed to Link to. Btw, why is 'link to' better than href? 'Link to' wil attach url prefix 'http://localhost:3000' to the link you set up. For example, if I change the code below to link to, the url will become 'http://localhost:3000/http://localhost:5000/images/07cea77f-a2ff-422f-ab35-3e918a025262.jpeg', which is not the route we want to have to show image.
```
Comment (EllieChen-Git) [inline src/components/images_uploading/DisplayImageList.js:98]: See my comment in code here: This one can't be changed to 'Link to'

File: src/components/images_uploading/UploadImageFile.js [hunk 10]
```diff
@@ -14,7 +13,9 @@ class UploadImageFile extends Component {
   };
 
   fileUploadHandler = event => {
-    event.preventDefault();
```
Comment (EllieChen-Git) [inline src/components/images_uploading/UploadImageFile.js:17]: The only way I found to refresh the page once we create a new apartment lol
I know this is definitely not the best way to do it, but at least it gets the work done XD
Will see if I can find a better way later :)

File: src/components/images_uploading/UploadImageFile.js [hunk 11]
```diff
@@ -31,6 +32,14 @@ class UploadImageFile extends Component {
       LocalApi.post("/images", formData).then(res => {
         console.log(res.statusText);
       });
+
+      //clear input field (reset to original state) after submitting the form
```
Comment (EllieChen-Git) [inline src/components/images_uploading/UploadImageFile.js:36]: After we create a new apartment, the data entered into the input fields will be cleared.

File: src/components/shared/NavBar.js [hunk 12]
```diff
@@ -4,12 +4,12 @@ import 'bootstrap/dist/css/bootstrap.min.css';
 
 class NavBar extends Component {
     render() {
```
Comment (EllieChen-Git) [inline src/components/shared/NavBar.js:6]: Nothing changed. You can ignore this file!
I tried to make a navbar, but our routes in App.js to too complicated haha.

File: src/App.js (hunk 2 above)
Comment (INRokh) [reply src/App.js:62]: thanks

File: src/App.js (hunk 3 above)
Comment (INRokh) [reply src/App.js:80]: thanks

File: src/components/images_uploading/DisplayImageList.js (hunk 9 above)
Comment (INRokh) [reply src/components/images_uploading/DisplayImageList.js:98]: That makes sense for now. When the image editing component is ready we can replace this direct link to backend with a Link to image editing route.

File: src/components/images_uploading/UploadImageFile.js (hunk 11 above)
Comment (INRokh) [reply src/components/images_uploading/UploadImageFile.js:36]: It is better to clear the state when we know that data is successfully uploaded:
LocalApi.post(...).then(res => {
  ...
  this.setState({
        selectedFile: null,
        lot: 0,
        unitNumber: "",
        productDescription: ""
      });
}).

**Review by INRokh — APPROVED** [review]
Awesome, thanks a lot!

File: src/components/images_uploading/DisplayImageList.js (hunk 9 above)
Comment (EllieChen-Git) [reply src/components/images_uploading/DisplayImageList.js:98]: Got it! Thanks :)

File: src/components/images_uploading/UploadImageFile.js (hunk 11 above)
Comment (EllieChen-Git) [reply src/components/images_uploading/UploadImageFile.js:36]: Cool! I'll amend the code accordingly :)

**Comment by EllieChen-Git** [comment]
Hi Nina, 
I'm working on 'editing apartments' today (hope I can make it happen lol). 
Let me know if you want me to leave this pull request here (without merging) until I implement 'editing apartments'  OR if you want to merge it now? I'm not quite sure which one is better as I might still change some code (just wanted to have you review this before I continue haha)
Cheers,
Ellie

**Comment by INRokh** [comment]
let's merge

**Pull request merged**